contracts/interface/IVaultFactory.sol

```diff
@@ -0,0 +1,11 @@
+// SPDX-License-Identifier: MIT
```

Comment:
We have to import the interface one way or another since the Vault has no way of knowing what functions the VaultFactory has. Will move this interface declaration to the Vault file instead of having another IVaultFactory file, since we only need the one function (importing the entire VaultFactory file wastes space).

The implementation we use for `hasRole` is still the one inherited from `AccessControl`